File: bot/src/main/java/net/slimediamond/atom/irc/IRCCommand.java
package net.slimediamond.atom.irc;

import net.slimediamond.atom.irc.annotations.Command;

import java.lang.reflect.Method;

public class IRCCommand {

    Command command;
    Object clazz;
    Method method;

    public IRCCommand(Command command, Object clazz, Method method) {
        this.command = command;
        this.clazz = clazz;
        this.method = method;
    }

    public Command getCommand() {
        return command;
    }

    public Object getCaller() {
        return clazz;
    }

    public Method getMethod() {
        return method;
    }
}

File: bot/src/main/kotlin/net/slimediamond/atom/commands/irc/IrcCommand.kt
package net.slimediamond.atom.commands.irc

import net.slimediamond.atom.api.command.RootOnlyCommandNode
import net.slimediamond.atom.api.command.platforms.CommandPlatforms

class IrcCommand : RootOnlyCommandNode("User-facing IRC commands", "irc") {

    init {
        platforms.add(CommandPlatforms.DISCORD)

        addChild(WhoisCommand())
    }

}